web/src/components/apps/Dashboard.jsx

```diff
@@ -159,6 +158,10 @@ class Dashboard extends Component {
 
   getAppDashboard = () => {
     return new Promise((resolve, reject) => {
+      if (this.props.cluster?.id == "" && this.props.isHelmManaged == true){
+        this.props.cluster.id = 0;
```

Comment:
```suggestion
        // TODO: use a callback to update the state in the parent component 
        this.props.cluster.id = 0;
```
Props should be immutable and updating them directly can cause side effects. I don't think it will cause any issues with this but in general the flow of data should be parent -> child component to prompt the correct rendering behavior. 

children communicate changes with callbacks (`this.props.clusterIdChanged(0)`), parents can handle those callback by updating their state.